src/main/java/com/uber/cadence/samples/hello/HelloCancellation.java

```diff
@@ -0,0 +1,149 @@
+     * activity invocations.
+     */
+    private final GreetingActivities activities =
+        Workflow.newActivityStub(GreetingActivities.class);
+
+    @Override
+    public String getGreeting(String name) throws Exception {
+      try {
+        Workflow.sleep(Duration.ofSeconds(10));
+        return activities.composeGreeting("Hello", name);
+      } catch (
+          CancellationException
+              e) { // This exception is thrown when a cancellation is requested on the current
+        // workflow
+        /**
+         * Any call to an activity or a child workflow after the workflow is cancelled is going to
+         * fail immediately with the CancellationException. the DetachedCancellationScope doesn't
+         * inherit its cancellation status from the enclosing scope. Thus it allows running a
+         * cleanup activity even if the workflow cancellation was requested.
+         */
+        Workflow.newDetachedCancellationScope(() -> activities.sayGoodbye(name));
+      }
+      return "Cancellation issued.. Goodbye.";
+    }
+  }
+
+  static class GreetingActivitiesImpl implements GreetingActivities {
+
+    private final List<String> invocations = new ArrayList<>();
+
+    @Override
+    public String composeGreeting(String greeting, String name) throws Exception {
+      invocations.add("composeGreeting");
+      return greeting + " " + name + "!";
+    }
+
+    @Override
+    public String sayGoodbye(String name) {
+      invocations.add("sayGoodbye");
+      return "Goodbye " + name + "!";
+    }
+
+    List<String> getInvocations() {
+      return invocations;
+    }
+  }
+
+  public static void main(String[] args) {
+    // Start a worker that hosts both workflow and activity implementations.
+    Worker.Factory factory = new Worker.Factory(DOMAIN);
+    Worker worker = factory.newWorker(TASK_LIST);
+
+    // Workflows are stateful. So you need a type to create instances.
+    worker.registerWorkflowImplementationTypes(GreetingWorkflowImpl.class);
+
+    // A shared instance is used to show activity invocations.
+    GreetingActivitiesImpl activities = new GreetingActivitiesImpl();
+    worker.registerActivitiesImplementations(activities);
+
+    // Start listening to the workflow and activity task lists.
+    factory.start();
+
+    // Start a workflow execution. Usually this is done from another program.
+    WorkflowClient workflowClient = WorkflowClient.newInstance(DOMAIN);
+    WorkflowOptions workflowOptions =
+        new WorkflowOptions.Builder()
+            .setTaskList(HelloCancellation.TASK_LIST)
+            .setExecutionStartToCloseTimeout(Duration.ofDays(30))
+            .build();
+    WorkflowStub client =
+        workflowClient.newUntypedWorkflowStub("GreetingWorkflow::getGreeting", workflowOptions);
+
+    client.start("World");
+
+    // issue cancellation request. This will trigger a CancellationException on the workflow.
+    client.cancel();
+
+    try {
+      client.getResult(String.class);
```

Comment:
Add throw new exception here to show that cancellation is thrown

Reply:
as in print that an exception was thrown or actually throw an exception here?